spec/Section 2 -- Language.md

```diff
@@ -938,3 +943,239 @@ and operations.
 
 As future versions of GraphQL adopts new configurable execution capabilities,
 they may be exposed via directives.
+
+
+## Type System Definition
+
+TypeSystemDefinition :
+  - SchemaDefinition
+  - TypeDefinition
+  - TypeExtensionDefinition
+  - DirectiveDefinition
+
+The GraphQL language also includes an
+[IDL](https://en.wikipedia.org/wiki/Interface_description_language) used to
+describe a GraphQL service's Type System. Tools may use these definitions to
+provide various utilities such as client code generation or
+service boot-strapping.
+
+GraphQL services which only seek to provide GraphQL query execution may
+choose to not parse {TypeSystemDefinition}.
+
+A GraphQL Document which contains {TypeSystemDefinition} must not be executed;
+GraphQL execution services which receive a GraphQL Document containing type
+system definitions should return a descriptive error.
+
+Note: This IDL is used throughout the remainder of this specification document
+when illustrating example type systems.
+
+
+### Schema Definition
+
+SchemaDefinition : schema { OperationTypeDefinition+ }
+
+OperationTypeDefinition : OperationType : NamedType
+
+A GraphQL Type System includes exactly one Schema Definition, which defines
+the type to be used for each operation.
+
+In this example, a GraphQL schema is defined with both query and mutation types:
+
+```graphql
+schema {
+  query: QueryRoot
+  mutation: MutationRoot
+}
+
+type QueryRoot {
+  someField: String
+}
+
+type MutationRoot {
+  setSomeField(to: String): String
+}
+```
+
+
+### Type Definition
+
+TypeDefinition :
+  - ScalarTypeDefinition
+  - ObjectTypeDefinition
+  - InterfaceTypeDefinition
+  - UnionTypeDefinition
+  - EnumTypeDefinition
+  - InputObjectTypeDefinition
+
+A GraphQL Type System is defined by many different kinds of types.
+
+
+#### Scalar
+
+ScalarTypeDefinition : scalar Name
+
+Scalar types represent leaf values in a GraphQL type system. While this GraphQL
+specification describes a set of Scalar types which all GraphQL services must
+supply, custom Scalar types may be supplied by a GraphQL service. Typically, the
+set of Scalar types described in this specification are omitted from documents
+which describe a schema for brevity.
+
+In this example, a Scalar type called `DateTime` is defined:
+
+```graphql
+scalar DateTime
+```
+
+
+#### Object
+
+ObjectTypeDefinition : type Name ImplementsInterfaces? { FieldDefinition+ }
```

Comment:
Thanks for the catch and close read. I'll take a closer look and figure out which is more appropriate, but I think you're probably right.